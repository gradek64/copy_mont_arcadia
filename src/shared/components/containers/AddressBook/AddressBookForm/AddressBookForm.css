/** @define AddressBookForm */

.AddressBookForm {
  &-close {
    float: right;
     font-size: 12px;
     letter-spacing: 1.5px;
     border-bottom: 1px solid #363636;
     padding-bottom: 4px;
     height: 23px;
     text-decoration: none;
  }

  &-primary {
    display: block;
    margin-bottom: 15px;
  }
}

@media $min-tablet {
  .AddressBookForm {
    &-primary {
      display: inline-block;
      width:auto;
      min-width:320px;
    }
  }
}

@media $min-laptop {
  .AddressBookForm {
    width:60%;
  }
}

