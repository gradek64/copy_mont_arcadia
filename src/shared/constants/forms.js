export const formNames = {
  deliveryCheckout: {
    address: 'yourAddress',
    details: 'yourDetails',
    findAddress: 'findAddress',
  },
  deliveryMCD: {
    address: 'deliveryAddressMCD',
    details: 'deliveryDetailsAddressMCD',
    findAddress: 'deliveryFindAddressMCD',
  },
  billingCheckout: {
    address: 'billingAddress',
    details: 'billingDetails',
    findAddress: 'billingFindAddress',
  },
  billingMCD: {
    address: 'billingAddressMCD',
    details: 'billingDetailsAddressMCD',
    findAddress: 'billingFindAddressMCD',
  },
  payment: {
    paymentCardDetailsMCD: 'paymentCardDetailsMCD',
    paymentCardDetails: 'billingCardDetails',
  },
  addressBook: {
    address: 'newAddress',
    details: 'newDetails',
    findAddress: 'newFindAddress',
    deliverToAddress: 'newDeliverToAddress',
  },
}

export const maxLengthName = 60

// Regex below checks that a string does not match any of those special characters
export const specialChars =
  '^((?![\u00A0\u1680â€�‹\u180e\u2000-\u2009\u200aâ€‹\u200bâ€‹\u202f\u205fâ€‹\u3000\u00b0°\u2013–\u2014—\u2022•\u005c\\\u00a2¢\u20a9₩\u00a5¥\u00a7§\u00ab«\u00bb»\u201e„\u201c“\u201d”\u2026…\u00bf¿\u00a1¡\u2018‘\u2019’]).)*$'
