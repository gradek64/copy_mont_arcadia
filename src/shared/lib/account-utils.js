import { WCS_ERRORS } from '../../shared/lib/wcsCodes'

/**
 * Utility function responsible to specifically map same password message by checking the errorCode
 * It returns a boolean value
 * @param errorCode
 */
export const passwordMatchesPrevious = (errorCode) =>
  errorCode === WCS_ERRORS.SAME_PASSWORD_RESET
