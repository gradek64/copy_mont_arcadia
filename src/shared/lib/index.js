export { PrivacyGuard } from './privacy-guard'
export { Measure } from './measure'
