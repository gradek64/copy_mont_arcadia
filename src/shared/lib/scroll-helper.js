let initY
let initDelta
let currY
let currDelta
let duration
let destination
let component
let scrollFunction
let directionIncrease

const scrollToY = (step) => {
  let nextStep

  if (!step) {
    nextStep = window.scrollY || window.pageYOffset
  } else if (
    (directionIncrease && step >= destination) ||
    (!directionIncrease && step <= destination)
  ) {
    window.scrollTo(0, destination)
    nextStep = destination
  } else {
    window.scrollTo(0, step)
    nextStep = step
  }

  return nextStep
}

const scrollComponent = (step) => {
  let nextStep

  if (!step) {
    nextStep = component.scrollTop
  } else if (
    (directionIncrease && step >= destination) ||
    (!directionIncrease && step <= destination)
  ) {
    component.scrollTop = destination
    nextStep = destination
  } else {
    component.scrollTop = step
    nextStep = step
  }

  return nextStep
}

const getDelta = (speed, curr, total) => {
  if (total === 0) return 0

  const method = Math.sin
  const PI = Math.PI
  const INIT_SPEED = 10

  return Math.abs(speed * method(PI * Math.abs(total - curr)) + INIT_SPEED)
}

const runScroll = (resolve) => {
  currDelta = getDelta(initDelta, currY, Math.abs(destination - initY))

  const newY = directionIncrease ? currY + currDelta : currY - currDelta

  if (newY === currY) return

  currY = newY

  if (scrollFunction(currY) !== destination) {
    window.requestAnimationFrame(() => runScroll(resolve))
  } else {
    resolve()
  }
}

const init = (dest, time, func, parent) => {
  return new Promise((resolve) => {
    destination = dest || 0
    duration = time || 1000
    component = parent || null
    scrollFunction = func

    initY = scrollFunction()
    directionIncrease = initY < destination

    initDelta = Math.abs(destination - initY) / (duration / 60)
    currY = initY

    runScroll(resolve)
  })
}

// scroll to the top of the page
export function scrollToTop(time) {
  return init(0, time, scrollToY)
}

// scroll to a point other than the top
export function scrollToPoint(dest, time) {
  return init(dest, time, scrollToY)
}

// scroll an element to the top of the viewport
// [optional] offset parameter which is the distance the element should end up from the top
export function scrollElementIntoView(element, time, offset = 0) {
  const getYCoord = (el, y) => {
    if (el === null) return y - offset
    return getYCoord(el.offsetParent, y + el.offsetTop)
  }
  return init(getYCoord(element, 0), time, scrollToY)
}

// scroll within a container
export function scrollWithinContainer(dest, time, parent) {
  return init(dest, time, scrollComponent, parent)
}

export const scrollToFormField = (fieldName) => {
  const el = document.querySelector(`.FormComponent-${fieldName}`)
  setTimeout(() => {
    scrollElementIntoView(el, 400, 20).then(() => {
      const field = el.querySelector('input, select, button')
      if (field) field.focus()
    })
  }, 15)
}

export const scrollToTopImmediately = () => {
  window.scrollTo(0, 0)
}
