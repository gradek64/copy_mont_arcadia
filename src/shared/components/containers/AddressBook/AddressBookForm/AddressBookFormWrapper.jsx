import { connect } from 'react-redux'
import AddressBookForm from './AddressBookForm'
import {
  getHasSelectedAddress,
  getHasFoundAddress,
} from '../../../../selectors/addressBookSelectors'
import {
  getFormNames,
  getIsFindAddressVisible,
  getCountryFor,
  getAddressForm,
} from '../../../../selectors/formsSelectors'
import {
  getYourDetailsSchema,
  getYourAddressSchema,
  getFindAddressSchema,
} from '../../../../schemas/validation/addressFormValidationSchema'
import {
  getPostCodeRules,
  getCountriesByAddressType,
} from '../../../../selectors/common/configSelectors'
import {
  touchedFormField,
  validateForm,
  resetForm,
  clearFormErrors,
  clearFormFieldError,
} from '../../../../actions/common/formActions'
import { validateDDPForCountry } from '../../../../actions/common/ddpActions'
import { bagContainsDDPProduct } from '../../../../selectors/shoppingBagSelectors'

export const mapStateToProps = (
  state,
  { addressType, onSaveAddress, onClose, closeAccordionWithAddress }
) => {
  const formNames = getFormNames(addressType)
  const addressForm = getAddressForm(addressType, formNames.address, state)
  const bagHasDDPProduct = bagContainsDDPProduct(state)

  const deliverToAddressForm = getAddressForm(
    addressType,
    formNames.deliverToAddress,
    state
  )
  const detailsForm = getAddressForm(addressType, formNames.details, state)
  const findAddressForm = getAddressForm(
    addressType,
    formNames.findAddress,
    state
  )
  const country = getCountryFor(addressType, formNames.address, state)
  const countries = getCountriesByAddressType(state, addressType)
  const postCodeRules = getPostCodeRules(state, country)
  const detailsValidationSchema = getYourDetailsSchema(country)
  const addressValidationSchema = getIsFindAddressVisible(
    addressType,
    formNames.address,
    country,
    state
  )
    ? {}
    : getYourAddressSchema(postCodeRules, countries)
  const findAddressValidationSchema = getIsFindAddressVisible(
    addressType,
    formNames.address,
    country,
    state
  )
    ? getFindAddressSchema(postCodeRules, {
        hasFoundAddress: getHasFoundAddress(findAddressForm),
        hasSelectedAddress: getHasSelectedAddress(addressForm),
      })
    : {}
  return {
    addressType,
    onSaveAddress,
    country,
    bagHasDDPProduct,
    onClose,
    addressForm,
    detailsForm,
    findAddressForm,
    deliverToAddressForm,
    addressValidationSchema,
    detailsValidationSchema,
    findAddressValidationSchema,
    closeAccordionWithAddress,
  }
}
export const mapDispatchToProps = {
  touchedFormField,
  validateForm,
  resetForm,
  clearFormErrors,
  clearFormFieldError,
  validateDDPForCountry,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddressBookForm)
