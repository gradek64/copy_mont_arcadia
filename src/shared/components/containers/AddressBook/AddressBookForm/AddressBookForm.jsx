import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AddressForm from '../../../common/AddressForm/AddressForm'
import DeliverToAddressBttnForm from '../../../common/AddressForm/DeliverToAddressBttnForm'
import AddressFormDetails from '../../../common/AddressFormDetails/AddressFormDetails'

import { getFormNames } from '../../../../selectors/formsSelectors'
import { pluck, isEmpty } from 'ramda'
import { validate } from '../../../../lib/validator/index'
import { scrollElementIntoView } from '../../../../lib/scroll-helper'
import { removeSpacing } from '../../../../lib/string-utils'
import Form from '../../../common/FormComponents/Form/Form'

export default class AddressBookForm extends Component {
  static contextTypes = {
    l: PropTypes.func,
  }
  static propTypes = {
    addressType: PropTypes.string.isRequired,
    onSaveAddress: PropTypes.func.isRequired,
    country: PropTypes.string.isRequired,
    bagHasDDPProduct: PropTypes.bool,
    onClose: PropTypes.func.isRequired,
    addressForm: PropTypes.object.isRequired,
    detailsForm: PropTypes.object.isRequired,
    findAddressForm: PropTypes.object.isRequired,
    addressValidationSchema: PropTypes.object.isRequired,
    detailsValidationSchema: PropTypes.object.isRequired,
    findAddressValidationSchema: PropTypes.object.isRequired,
    touchedFormField: PropTypes.func.isRequired,
    closeAccordionWithAddress: PropTypes.func.isRequired,
    clearFormFieldError: PropTypes.func.isRequired,
    validateDDPForCountry: PropTypes.func.isRequired,
  }

  static defaultProps = {
    bagHasDDPProduct: false,
  }

  scrollToFirstError = (name) => {
    if (process.browser) {
      const el = document.querySelector(
        `.AddressBookForm .FormComponent-${name}`
      )
      scrollElementIntoView(el, 400, 20)
    }
  }
  validateForm = (validationSchema, addressForm, formName) => {
    const { l } = this.context
    const { touchedFormField } = this.props
    const errors = validate(validationSchema, addressForm, l)
    if (!isEmpty(errors)) {
      this.scrollToFirstError(Object.keys(errors)[0])
      Object.keys(validationSchema).forEach((name) =>
        touchedFormField(formName, name)
      )
      return false
    }
    return true
  }

  onSubmitForm = (event) => {
    event.preventDefault()
    const {
      onSaveAddress,
      addressType,
      addressForm,
      detailsForm,
      findAddressForm,
      addressValidationSchema,
      detailsValidationSchema,
      findAddressValidationSchema,
      clearFormFieldError,
      country,
      bagHasDDPProduct,
      validateDDPForCountry,
      closeAccordionWithAddress,
    } = this.props
    const formNames = getFormNames(addressType)
    const address = pluck('value', addressForm.fields)
    const details = pluck('value', detailsForm.fields)
    const findAddress = pluck('value', findAddressForm.fields)

    clearFormFieldError(formNames.deliverToAddress, 'deliverToAddress')
    const isFormValid = [
      this.validateForm(detailsValidationSchema, details, formNames.details),
      this.validateForm(
        findAddressValidationSchema,
        findAddress,
        formNames.findAddress
      ),
      this.validateForm(addressValidationSchema, address, formNames.address),
    ].every((value) => value)
    if (isFormValid) {
      onSaveAddress({
        address,
        nameAndPhone: {
          ...details,
          telephone: removeSpacing(details.telephone),
        },
      }).then(() => {
        // inform customer that his DDP subscription doesn`t work outside UK
        if (bagHasDDPProduct) validateDDPForCountry(country)
        // close accordion with selected address
        closeAccordionWithAddress()
      })
    }
  }

  render() {
    const { l } = this.context
    const {
      addressType,
      onClose,
      detailsForm,
      addressForm,
      findAddressForm,
      deliverToAddressForm,
      validateForm,
      resetForm,
      clearFormErrors,
    } = this.props

    return (
      <div className="AddressBookFormWrapper">
        <a
          href=""
          className="AddressBookForm-close"
          onClick={(event) => {
            event.preventDefault()
            onClose()
          }}
        >
          {l`CANCEL`}
        </a>
        <Form className="AddressBookForm">
          <AddressForm addressType={addressType} />
          <AddressFormDetails addressType={addressType} />
          <DeliverToAddressBttnForm
            forms={{
              detailsForm,
              findAddressForm,
              addressForm,
              deliverToAddressForm,
            }}
            addressType={addressType}
            onSubmitForm={this.onSubmitForm}
            validateForm={validateForm}
            resetForm={resetForm}
            clearFormErrors={clearFormErrors}
          />
        </Form>
      </div>
    )
  }
}
