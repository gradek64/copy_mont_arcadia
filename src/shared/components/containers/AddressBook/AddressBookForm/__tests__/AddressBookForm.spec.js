import { compose } from 'ramda'
import testComponentHelper, {
  buildComponentRender,
  shallowRender,
  mountRender,
  withStore,
} from '../../../../../../../test/unit/helpers/test-component'
import AddressBookForm from '../AddressBookForm'
import { addressBookFormMock } from '../../../../../../../test/mocks/forms/addressBookFormMock'
import {
  getPostCodeRules,
  getCountriesByAddressType,
} from '../../../../../selectors/common/configSelectors'
import {
  getYourAddressSchema,
  getYourDetailsSchema,
  getFindAddressSchema,
} from '../../../../../schemas/validation/addressFormValidationSchema'
import { scrollElementIntoView } from '../../../../../lib/scroll-helper'

jest.mock('../../../../../lib/scroll-helper', () => ({
  scrollElementIntoView: jest.fn(),
}))

describe('AddressBookForm', () => {
  const state = {
    forms: {
      addressBook: addressBookFormMock,
    },
    features: {
      status: {
        FEATURE_ADDRESS_BOOK: true,
      },
    },
    config: {
      deliveryCountries: [{ iso: 'UK', name: 'United Kingdom' }],
      checkoutAddressFormRules: {
        'United Kingdom': {
          pattern:
            '^(([gG][iI][rR] {0,}0[aA]{2})|(([aA][sS][cC][nN]|[sS][tT][hH][lL]|[tT][dD][cC][uU]|[bB][bB][nN][dD]|[bB][iI][qQ][qQ]|[fF][iI][qQ][qQ]|[pP][cC][rR][nN]|[sS][iI][qQ][qQ]|[iT][kK][cC][aA]) {0,}1[zZ]{2})|((([a-pr-uwyzA-PR-UWYZ][a-hk-yxA-HK-XY]?[0-9][0-9]?)|(([a-pr-uwyzA-PR-UWYZ][0-9][a-hjkstuwA-HJKSTUW])|([a-pr-uwyzA-PR-UWYZ][a-hk-yA-HK-Y][0-9][abehmnprv-yABEHMNPRV-Y]))) {0,}[0-9][abd-hjlnp-uw-zABD-HJLNP-UW-Z]{2}))$',
          postcodeLabel: 'Postcode',
          premisesRequired: false,
          premisesLabel: 'House number',
        },
      },
    },
    siteOptions: {
      titles: ['Mr', 'Mrs', 'Ms', 'Miss', 'Dr'],
    },
  }
  const countries = getCountriesByAddressType(state, 'addressBook')
  const rules = getPostCodeRules(state, 'United Kingdom')
  const renderComponent = testComponentHelper(AddressBookForm)
  describe('@render', () => {
    it('should show the newAddress form', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(),
        closeAccordionWithAddress: jest.fn(),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '',
            },
            address2: {
              value: null,
            },
            city: {
              value: '',
            },
            country: {
              value: '',
            },
            postcode: {
              value: '',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: '',
            },
            firstName: {
              value: '',
            },
            lastName: {
              value: '',
            },
            telephone: {
              value: '',
            },
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: {},
        detailsValidationSchema: {},
        findAddressValidationSchema: {},
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const { getTree } = renderComponent(props)
      expect(getTree()).toMatchSnapshot()
    })
  })
  describe('@events', () => {
    it('should call the onSaveAddress function when the form is submitted with a valid address', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(() => Promise.resolve()),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '1 Acme Road',
            },
            address2: {
              value: null,
            },
            city: {
              value: 'London',
            },
            country: {
              value: 'United Kingdom',
            },
            postcode: {
              value: 'EC3N 1JJ',
            },
            state: {
              value: '',
            },
            deliverToAddress: '',
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: 'Mr',
            },
            firstName: {
              value: 'Jose',
            },
            lastName: {
              value: 'Quinto',
            },
            telephone: {
              value: '01234567890',
            },
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules, countries),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom', {
          hasSelectedAddress: true,
        }),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddresses: true,
          hasSelectedAddress: true,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalledWith(
        'newDeliverToAddress',
        'deliverToAddress'
      )
      expect(props.onSaveAddress).toHaveBeenCalledTimes(1)
    })
    it('should call the onSaveAddress function when the form is submitted with a valid address & phone number that has a space', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(() => Promise.resolve()),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '1 Acme Road',
            },
            address2: {
              value: null,
            },
            city: {
              value: 'London',
            },
            country: {
              value: 'United Kingdom',
            },
            postcode: {
              value: 'EC3N 1JJ',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: 'Mr',
            },
            firstName: {
              value: 'Jose',
            },
            lastName: {
              value: 'Quinto',
            },
            telephone: {
              value: '01234 567890',
            },
            deliverToAddress: '',
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules, countries),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom', {
          hasSelectedAddress: true,
        }),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddresses: true,
          hasSelectedAddress: true,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalledWith(
        'newDeliverToAddress',
        'deliverToAddress'
      )
      expect(props.onSaveAddress).toHaveBeenCalledTimes(1)
    })
    it('should call the validateDDPForCountry on onSaveAddress success when bag has DDP product', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(() => Promise.resolve()),
        closeAccordionWithAddress: jest.fn(),
        bagHasDDPProduct: false,
        validateDDPForCountry: jest.fn(),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '1 Acme Road',
            },
            address2: {
              value: null,
            },
            city: {
              value: 'London',
            },
            country: {
              value: 'United Kingdom',
            },
            postcode: {
              value: 'EC3N 1JJ',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: 'Mr',
            },
            firstName: {
              value: 'Jose',
            },
            lastName: {
              value: 'Quinto',
            },
            telephone: {
              value: '01234 567890',
            },
            deliverToAddress: '',
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules, countries),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom', {
          hasSelectedAddress: true,
        }),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddresses: true,
          hasSelectedAddress: true,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalledWith(
        'newDeliverToAddress',
        'deliverToAddress'
      )
      props.onSaveAddress().then(() => {
        expect(props.validateDDPForCountry).toHaveBeenCalled()
        expect(props.closeAccordionWithAddress).toHaveBeenCalled()
      })
    })
    it('should not call the onSaveAddress function when the form is submitted with no details', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '1 Acme Road',
            },
            address2: {
              value: null,
            },
            city: {
              value: 'London',
            },
            country: {
              value: 'United Kingdom',
            },
            postcode: {
              value: 'EC3N 1JJ',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: '',
            },
            firstName: {
              value: '',
            },
            lastName: {
              value: '',
            },
            telephone: {
              value: '',
            },
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules, countries),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom'),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddress: true,
          hasSelectedAddress: true,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const render = compose(
        shallowRender,
        withStore(state)
      )
      const renderComponent = buildComponentRender(render, AddressBookForm)
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalledWith(
        'newDeliverToAddress',
        'deliverToAddress'
      )
      expect(props.onSaveAddress).not.toHaveBeenCalled()
    })
    it('should not call the onSaveAddress function when the form is submitted with no address', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '',
            },
            address2: {
              value: null,
            },
            city: {
              value: '',
            },
            country: {
              value: '',
            },
            postcode: {
              value: '',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: 'Mr',
            },
            firstName: {
              value: 'Jose',
            },
            lastName: {
              value: 'Quinto',
            },
            telephone: {
              value: '01234567890',
            },
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom'),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddress: true,
          hasSelectedAddress: true,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const render = compose(
        shallowRender,
        withStore(state)
      )
      const renderComponent = buildComponentRender(render, AddressBookForm)
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalledWith(
        'newDeliverToAddress',
        'deliverToAddress'
      )
      expect(props.onSaveAddress).not.toHaveBeenCalled()
    })
    it('should not call the onSaveAddress function when the form is submitted with details and no find address', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(),
        onClose: jest.fn(),
        addressForm: {
          fields: {
            address1: {
              value: '',
            },
            address2: {
              value: null,
            },
            city: {
              value: '',
            },
            country: {
              value: '',
            },
            postcode: {
              value: '',
            },
            state: {
              value: '',
            },
          },
          errors: {},
        },
        detailsForm: {
          fields: {
            title: {
              value: 'Mr',
            },
            firstName: {
              value: 'Jose',
            },
            lastName: {
              value: 'Quinto',
            },
            telephone: {
              value: '01234567890',
            },
          },
          errors: {},
        },
        findAddressForm: {
          fields: {
            houseNumber: {
              value: '',
            },
            postcode: {
              value: '',
            },
          },
          errors: {},
        },
        addressValidationSchema: getYourAddressSchema(rules, countries),
        detailsValidationSchema: getYourDetailsSchema('United Kingdom'),
        findAddressValidationSchema: getFindAddressSchema(rules, {
          hasFoundAddress: false,
          hasSelectedAddress: false,
        }),
        touchedFormField: jest.fn(),
        clearFormFieldError: jest.fn(),
        dispatch: jest.fn(),
      }
      const render = compose(
        shallowRender,
        withStore(state)
      )
      const renderComponent = buildComponentRender(render, AddressBookForm)
      const { wrapper } = renderComponent(props)
      wrapper.find('DeliverToAddressBttnForm').prop('onSubmitForm')({
        preventDefault: jest.fn(),
      })
      expect(props.clearFormFieldError).toHaveBeenCalled()
      expect(props.onSaveAddress).not.toHaveBeenCalled()
    })
    it('should call onClose when the close link is clicked', () => {
      const props = {
        addressType: 'addressBook',
        onSaveAddress: jest.fn(),
        onClose: jest.fn(),
        addressForm: addressBookFormMock.newAddress,
        detailsForm: addressBookFormMock.newDetails,
        findAddressForm: addressBookFormMock.newFindAddress,
        addressValidationSchema: {},
        detailsValidationSchema: {},
        findAddressValidationSchema: {},
        clearFormFieldError: jest.fn(),
        touchedFormField: jest.fn(),
      }
      const { wrapper } = renderComponent(props)
      const closeLink = wrapper.find('.AddressBookForm-close').first()
      closeLink.simulate('click', { preventDefault: jest.fn() })
      expect(props.onClose).toHaveBeenCalledTimes(1)
    })
  })

  describe('@functions', () => {
    describe('scrollToFirstError', () => {
      it('should call scrollElementInView for address1 formComponent', () => {
        global.process.browser = true
        const props = {
          addressType: 'addressBook',
          onSaveAddress: jest.fn(),
          onClose: jest.fn(),
          addressForm: {
            fields: {
              address1: {
                value: '',
              },
              address2: {
                value: null,
              },
              city: {
                value: '',
              },
              country: {
                value: '',
              },
              postcode: {
                value: '',
              },
              state: {
                value: '',
              },
            },
            errors: {},
          },
          detailsForm: {
            fields: {
              title: {
                value: '',
              },
              firstName: {
                value: '',
              },
              lastName: {
                value: '',
              },
              telephone: {
                value: '',
              },
            },
            errors: {},
          },
          findAddressForm: {
            fields: {
              houseNumber: {
                value: '',
              },
              postcode: {
                value: '',
              },
            },
            errors: {},
          },
          addressValidationSchema: {},
          detailsValidationSchema: {},
          findAddressValidationSchema: {},
          touchedFormField: jest.fn(),
          dispatch: jest.fn(),
        }
        const render = compose(
          mountRender,
          withStore(state)
        )
        const renderComponent = buildComponentRender(render, AddressBookForm)
        const { instance } = renderComponent(props)
        jest.spyOn(document, 'querySelector').mockImplementation(() => {
          return 'address1'
        })
        instance.scrollToFirstError('address1')
        expect(scrollElementIntoView).toHaveBeenCalledWith('address1', 400, 20)
      })
    })
  })
})
