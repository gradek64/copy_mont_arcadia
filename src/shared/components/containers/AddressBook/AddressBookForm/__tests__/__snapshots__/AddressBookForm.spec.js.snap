// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBookForm @render should show the newAddress form 1`] = `
<div
  className="AddressBookFormWrapper"
>
  <a
    className="AddressBookForm-close"
    href=""
    onClick={[Function]}
  >
    CANCEL
  </a>
  <Form
    className="AddressBookForm"
    noValidate={true}
  >
    <Connect(AddressForm)
      addressType="addressBook"
    />
    <Connect(AddressFormDetails)
      addressType="addressBook"
    />
    <DeliverToAddressBttnForm
      addressType="addressBook"
      clearFormErrors={[Function]}
      deliverToAddressErrorMsg="Please confirm this new address or cancel to continue"
      forms={
        Object {
          "addressForm": Object {
            "errors": Object {},
            "fields": Object {
              "address1": Object {
                "value": "",
              },
              "address2": Object {
                "value": null,
              },
              "city": Object {
                "value": "",
              },
              "country": Object {
                "value": "",
              },
              "postcode": Object {
                "value": "",
              },
              "state": Object {
                "value": "",
              },
            },
          },
          "deliverToAddressForm": undefined,
          "detailsForm": Object {
            "errors": Object {},
            "fields": Object {
              "firstName": Object {
                "value": "",
              },
              "lastName": Object {
                "value": "",
              },
              "telephone": Object {
                "value": "",
              },
              "title": Object {
                "value": "",
              },
            },
          },
          "findAddressForm": Object {
            "errors": Object {},
            "fields": Object {
              "houseNumber": Object {
                "value": "",
              },
              "postcode": Object {
                "value": "",
              },
            },
          },
        }
      }
      onSubmitForm={[Function]}
      resetForm={[Function]}
      validateForm={[Function]}
    />
  </Form>
</div>
`;
